components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/factory/RegistrationRequestFactory.java

```diff
@@ -117,6 +119,13 @@ public void create(IdentityRequest.IdentityRequestBuilder builder, HttpServletRe
             if (obj != null && obj instanceof JSONArray) {
                 JSONArray redirectUris = (JSONArray) obj;
                 for (int i = 0; i < redirectUris.size(); i++) {
+                    try{
+                        URL redirectURL = new URL(redirectUris.get(i).toString());
```

Comment:
@Pushpalanka Need to test if the Array element is null (i.e. JSON null) before doing anything else. Some points of doing so is,
http://stackoverflow.com/questions/12251729/how-to-check-if-jsonarray-element-is-null